skyd/skyd.go

```diff
@@ -1,16 +1,137 @@
 package skyd
 
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"gitlab.com/NebulousLabs/errors"
+	"gitlab.com/SkynetLabs/skyd/node/api/client"
+	"gitlab.com/SkynetLabs/skyd/skymodules"
+)
+
+var (
+	// skylinksCache is a local cache of the list of skylinks pinned by the
+	// local skyd
+	skylinksCache pinnedSkylinksCache
+	// skylinksCacheDuration defines the duration of skylinksCache
+	skylinksCacheDuration = time.Hour
+)
+
 type (
 	// Client allows us to call the local skyd instance.
 	Client struct {
+		staticClient *client.Client
+	}
+	// ClientInterface describes the interface exposed by Client, so we can
+	// mock it for testing.
+	ClientInterface interface {
```

Comment:
This is the interface that allows us to pass a mock instead of the actual implementation. It needs to be here, unless we want to refer to the `test` package from the `scanner` package.

This is not test code, this code that accommodates and allows testing.

I'll remove the `so we can mock it for testing` from the docstring, so it doesn't cause confusion.

Reply:
Regarding the exposed vs unexposed implementation and returning interfaces. The general advice here has been to accept interfaces and return concrete types. The motivation being that anything that satisfies what we need in the function should be accepted (i.e. an interface) while we should return as much as possible (i.e. a concrete type) even if that has more methods than what's strictly required by the interface. It goes back the Robustness principle (https://en.wikipedia.org/wiki/Robustness_principle).